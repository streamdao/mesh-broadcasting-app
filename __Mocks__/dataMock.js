// eslint-disable-next-line import/prefer-default-export
const mockMessage1 = {
  fromUID: '0205f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd92b0be9c052a36ba5bc1d6',
  toUID: 'broadcast',
  messages: [],
  timestamp: 1580425894630,
  queue: [],
};

const mockMessage2 = {
  fromUID: '0205f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd92b0be9c052a36ba5bc1d6',
  toUID: '0202f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd91b0be9c052a30ba5bc1d9',
  messages: [],
  timestamp: 1580425894630,
  queue: [],
};

const mockContact1 = {
  name: 'test1',
  picture: null,
  uid: '02e5f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd92b0be9c052a36ba5bc1d6',
  hashUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1'
};


const mockContact2 = {
  name: 'test2',
  picture: null,
  uid: '02e5f8f594f8ca7e27d0f0b3e37432bcad4c066f4bbd91b0be9c052a36ba5bc1d6',
  hashUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1weq'
};

const mockUser = {
  uid: '02e5f8f594f8ca7e27d0f0b3e37432bcad4c066f4bbd91b0be9c052a36ba5bc1d6',
  peerID: '02e5f8f594f8ca7e27d0f0b3e37432bcad4c066f4bbd91b0be9c052a36ba5bc1d6',
  name: 'obj.name',
  image: null,
  contacts: [],
  chats: [
    {
      fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
      toUID: 'broadcast',
      messages: []
    }
  ]
};


const mocksetMessage = {
  fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  toUID: null,
  msgID: '5c28f23b375d47994b30190b01338ea18daa0b307909a2d465a597772469634e5',
  msg: {
    text: 'message'
  },
  timestamp: 12345512421242,
  type: 1
};


const mocksetMessage2 = {
  fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  toUID: null,
  msgID: '5c28f23b375d47994b30190b01338ea18daa0b307909a2d465a597772469634e512334123',
  msg: {
    text: 'message'
  },
  timestamp: 12345512421242,
  type: 1
};

const mocksetMessage3 = {
  fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  toUID: null,
  msgID: '5c28f23b375d47994b30190b01338ea18daa0b307909a2d465a5977724634123',
  msg: {
    text: 'message'
  },
  timestamp: 12345512421242,
  type: 1
};


const mocksetMessage4 = {
  fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  toUID: null,
  msgID: '5c28f23b375d47994b30190b01338ea18daa0b307909a2d465a59777246341231242',
  msg: {
    text: 'message x2'
  },
  timestamp: 12345512421242,
  type: 1
};

const mockContact = {
  uid: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  name: 'test',
  picture: null,
  hashUID: '0202f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd91b0be9c052a30ba5bc1d9'
};


const temporalInfo = {
  hashUID: '0202f8f594f8ca7e27d0f0b3e37430bcad4c066f4bbd91b0be9c052a30ba5bc1d923124sf124asf',
  name: 'temporal',
  timestamp: 1234251234
};


const temporalInfo2 = {
  hashUID: undefined,
  name: 'temporal',
  timestamp: 1234251234
};


const messageStatus = {
  fromUID: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
  toUID: null,
  timestamp: 1234251234,
  data: {
    status: 'delivered',
    msgID: '5c28f23b375d47994b30190b01338ea18daa0b307909a2d465a59777246341231242'
  },
  type: 2
};


export default {
  mockContact1,
  mockContact2,
  mockMessage2,
  mockMessage1,
  mocksetMessage,
  mocksetMessage2,
  mocksetMessage3,
  mocksetMessage4,
  mockUser,
  mockContact,
  temporalInfo,
  temporalInfo2,
  messageStatus
};
