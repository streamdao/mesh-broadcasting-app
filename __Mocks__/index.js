/* eslint-disable import/no-unresolved */
/* eslint-disable global-require */
/* eslint-disable react/prefer-stateless-function */
/* eslint-disable no-unused-vars */
/* eslint-disable max-classes-per-file */
/* eslint-disable import/no-extraneous-dependencies */
// eslint-disable-next-line no-unused-vars
import * as ReactNative from 'react-native';
import enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mock from 'react-native-permissions/mock';

// enzyme.configure({ adapter: new Adapter() });
jest.mock('react-native-fs', () => ({
  mkdir: jest.fn(),
  moveFile: jest.fn(() => new Promise((resolve) => {
    resolve(true);
  })),
  copyFile: jest.fn(),
  pathForBundle: jest.fn(),
  pathForGroup: jest.fn(),
  getFSInfo: jest.fn(),
  getAllExternalFilesDirs: jest.fn(),
  unlink: jest.fn(),
  exists: jest.fn(() => new Promise((resolve) => {
    resolve(true);
  })),
  stopDownload: jest.fn(),
  resumeDownload: jest.fn(),
  isResumable: jest.fn(),
  stopUpload: jest.fn(),
  completeHandlerIOS: jest.fn(),
  readDir: jest.fn(),
  readDirAssets: jest.fn(),
  existsAssets: jest.fn(),
  readdir: jest.fn(),
  setReadable: jest.fn(),
  stat: jest.fn(),
  readFile: jest.fn((path) => new Promise((resolve) => {
    resolve(path);
  })),
  read: jest.fn(),
  readFileAssets: jest.fn(),
  hash: jest.fn(),
  copyFileAssets: jest.fn(),
  copyFileAssetsIOS: jest.fn(),
  copyAssetsVideoIOS: jest.fn(),
  writeFile: jest.fn(),
  appendFile: jest.fn(),
  write: jest.fn(),
  downloadFile: jest.fn(),
  uploadFiles: jest.fn(),
  touch: jest.fn(),
  MainBundlePath: jest.fn(),
  CachesDirectoryPath: jest.fn(),
  DocumentDirectoryPath: jest.fn(),
  ExternalDirectoryPath: jest.fn(),
  ExternalStorageDirectoryPath: jest.fn(),
  TemporaryDirectoryPath: jest.fn(),
  LibraryDirectoryPath: jest.fn(),
  PicturesDirectoryPath: jest.fn(),
}));


jest.mock('react-native-share', () => {
  // code here
});


jest.mock('@react-native-community/datetimepicker', () => {
  // code here
});

// jest.mock('@react-native-picker/picker', () => {
//   // code here
// });

jest.mock('react-native-gesture-handler', () => {
  // code here
});


jest.mock('react-navigation-stack', () => {
  // code here
});


jest.mock('react-native-permissions', () => mock);


jest.mock('@react-native-community/slider', () => {
  const data = () => null;
  return data;
});


jest.mock('react-native-image-crop-picker', () => ({
  openPicker: jest.fn(() => new Promise((resolve) => {
    resolve({
      data: 'test',
      path: 'test'
    });
  })),
  openCamera: jest.fn(() => new Promise((resolve) => {
    resolve({
      data: 'test',
      path: 'test'
    });
  })),
}));


jest.mock('react-navigation', () => ({
  createAppContainer: jest.fn().mockReturnValue(() => null),
  createDrawerNavigator: jest.fn(),
  createMaterialTopTabNavigator: jest.fn(),
  createStackNavigator: jest.fn(),
  StackActions: {
    push: jest.fn().mockImplementation((x) => ({ ...x, type: 'Navigation/PUSH' })),
    replace: jest.fn().mockImplementation((x) => ({ ...x, type: 'Navigation/REPLACE' })),
  },
  NavigationActions: {
    navigate: jest.fn().mockImplementation((x) => x),
  }
}));

jest.mock('rn-fetch-blob', () => ({
  config: jest.fn().mockReturnValue(() => {
    jest.fn().mockReturnValue(() => new Promise());
  })
}));


jest.mock(
  '../node_modules/react-native/Libraries/EventEmitter/NativeEventEmitter',
);


jest.mock('native-base/dist/src/basic/Icon', () => jest.genMockFromModule('native-base/dist/src/basic/Icon'));


jest.mock('react-native/Libraries/Utilities/Platform', () => ({
  OS: 'ios',
  select: jest.fn((selector) => selector.ios),
}));

jest.mock('@react-native-community/async-storage', () => ({
  getItem: jest.fn((data) => new Promise((resolve) => {
    if (data === 'upnp') {
      resolve(true);
    } else {
      resolve(null);
    }
  })),

  setItem: jest.fn(() => new Promise((resolve) => {
    resolve(true);
  })),
  removeItem: jest.fn(() => new Promise((resolve) => {
    resolve();
  }))
}));


jest.mock('react-native-document-picker', () => ({
  pick: jest.fn(() => new Promise((resolve) => {
    resolve({
      uri: 'test'
    });
  })),
  types: {
    allFiles: true
  }
}));


jest.mock('react-native-exception-handler', () => ({
  setJSExceptionHandler: jest.fn((callback) => callback({
    message: 'test'
  }, true)),
  setNativeExceptionHandler: jest.fn((callback) => callback('data'))
}));


jest.doMock('react-native', () =>
  // Extend ReactNative
  Object.setPrototypeOf(
    {
      // Mock a native module
      NativeModules: {
        ...ReactNative.NativeModules,
        Override: { great: 'success' },
        LocalNotification: {
          requestPermission: jest.fn(),
        },
        PlayerModule: {
          release: jest.fn(),
          getCurrentTime: jest.fn((key, callback) => callback({ seconds: 0, isPlaying: false })),
          play: jest.fn(() => new Promise((resolve) => { resolve(true); })),
          setCurrentTime: jest.fn(),
          pause: jest.fn(() => new Promise((resolve) => { resolve(true); })),
          prepare: jest.fn(() => new Promise((resolve) => {
            resolve({
              duration: 0,
              key: 'test'
            });
          })),
        },
        SoundMdoule: {
          prepareRecoder: jest.fn(() => new Promise((resolve) => { resolve('test'); })),
          startRecording: jest.fn(),
          stopRecording: jest.fn()
        },
        RNDeviceInfo: {
          VersionInfo: '1',
          getIpv6Andipv4Adress: jest.fn().mockReturnValue(['192.168.0.1']),
          scanFile: jest.fn()
        },
        ChatService: {
          start: jest.fn(() => new Promise((resolve) => {
            resolve('5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1');
          })),
          stop: jest.fn(() => new Promise((resolve) => { resolve(); })),
          dial: jest.fn(),
          sendMessage: jest.fn(),
          addNewChatService: jest.fn((xpriv, address) => new Promise((resolve) => { resolve(); })),
        },
        bitcoinModule: {
          createWallet: jest.fn(() => new Promise((resolve) => {
            resolve({
              pubKey: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1',
              privKey: '5c28fab375d47994b30190b01338ea48daa0b307909a3d465a597772469633e1'
            });
          })),
          sha256: jest.fn(() => new Promise((resolve) => { resolve('test'); })),
          getPrivateKey: jest.fn(() => new Promise((resolve) => { resolve('test'); })),
          decrypt: jest.fn(() => new Promise((resolve) => {
            resolve('{"test":"test"}');
          })),
        },
        RNPermissions: {
          ...mock
        }
      },
      PermissionsAndroid: {
        check: jest.fn(() => new Promise((resolve) => { resolve(true); })),
        request: jest.fn((param) => new Promise((resolve) => { resolve(true); })),
        PERMISSIONS: {
          RECORD_AUDIO: true,
          RESULTS: {
            GRANTED: true
          }
        },
      },
      StyleSheet: {
        create: () => ({}),
        flatten: () => ({})
      },
      Clipboard: { setString: jest.fn() },
      Platform: {
        OS: 'ios',
        select: jest.fn((selector) => selector.ios),
      }
    },
    ReactNative,
  ));
