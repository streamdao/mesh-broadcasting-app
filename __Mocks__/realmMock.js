/* eslint-disable no-empty */
/* eslint-disable max-classes-per-file */
export default class Realm {
  constructor(params) {
    this.schema = {};
    this.data = {};
    this.schemaCallbackList = {};
    params.schema.forEach((schema) => {
      this.data[schema.name] = {};
    });
    params.schema.forEach((schema) => {
      this.schema[schema.name] = schema;
    });
    this.lastLookedUpModel = null;
  }

  objects(schemaName) {
    this.lastLookedUpModel = schemaName;
    const objects = Object.values(this.data[schemaName]);
    objects.values = () => objects;
    objects.sorted = () => (this.compareFunc ? objects.sort(this.compareFunc) : objects.sort());
    objects.addListener = (cb) => {
      if (this.schemaCallbackList[schemaName]) {
        this.schemaCallbackList[schemaName].push(cb);
      } else {
        this.schemaCallbackList[schemaName] = [cb];
      }
    };
    objects.removeListener = () => { };
    objects.filtered = this.filtered ? this.filtered.bind(this, schemaName) : () => objects;
    return objects;
  }

  write(fn) {
    this.writing = true;
    fn();
    this.writing = false;
  }


  verifySchema(id) {
    const exist = !!this.schema[id];
    return exist;
  }

  create(schemaName, object) {
    const modelObject = object;
    const { properties } = this.schema[schemaName];
    const data = {};
    Object.keys(properties).forEach((key) => {
      if (modelObject[key]) {
        data[key] = modelObject[key];
      } else if (!modelObject[key]) {
        if (typeof properties[key] === 'string') {
          if (properties[key].substring(properties[key].length - 1) === '?') {
            data[key] = null;
          } else {
            throw new Error(`${properties[key]} is required`);
          }
        } else {
          const resultSchema = this.verifySchema(properties[key].objectType);
          if (resultSchema && properties[key].type === 'list') {
            data[key] = {};
          }
        }
      }
    });
  }

  objectForPrimaryKey(model, id) {
    this.lastLookedUpModel = model;
    return this.data[model][id];
  }

  delete(object) {
    if (this.lastLookedUpModel || object.model) {
      const model = object.model ? object.model : this.lastLookedUpModel;
      if (Array.isArray(object)) {
        object.forEach((item) => {
          delete this.data[model][item.id];
        });
      }
      delete this.data[model][object.id];
      if (this.writing) {
        if (this.schemaCallbackList[model]) {
          this.schemaCallbackList[model].forEach((cb) => cb(model, {
            insertions: { length: 0 },
            modifications: { length: 0 },
            deletions: { length: 1 },
          }));
        }
        this.callbackList.forEach((cb) => { cb(); });
      }
    }
  }

  deleteAll() {
    Object.keys(this.schema).forEach((key) => {
      if (this.writing && this.schemaCallbackList[this.schema[key].name]) {
        this.schemaCallbackList[this.schema[key].name].forEach((cb) => cb(key, {
          insertions: { length: 0 },
          modifications: { length: 0 },
          deletions: { length: Object.values(this.data[this.schema[key].name]).length },
        }));
      }
      this.data[this.schema[key].name] = {};
    });
    if (this.writing) this.callbackList.forEach((cb) => { cb(); });
  }

  addListener(_event, callback) {
    this.callbackList.push(callback);
  }

  prepareData(schemaName, objects) {
    objects.forEach((object) => {
      this.create(schemaName, object);
    });
  }
}

Realm.Object = class Object {
  // eslint-disable-next-line class-methods-use-this
  isValid() { return true; }
};
