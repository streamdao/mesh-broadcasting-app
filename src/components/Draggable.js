/* eslint-disable no-return-assign */
/* eslint-disable no-underscore-dangle */
import React, { Component } from 'react';
import {
  PanResponder, Animated, View
} from 'react-native';

export default class Draggable extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showDraggable: true,
      // eslint-disable-next-line react/no-unused-state
      dropAreaValues: null,
      pan: new Animated.ValueXY(),
      // eslint-disable-next-line react/no-unused-state
      opacity: new Animated.Value(1)
    };
    this.init();
  }

  init() {
    this._val = { x: 0, y: 0 };
    this.state.pan.addListener((value) => (this._val = value));

    this.panResponder = PanResponder.create({
      onStartShouldSetPanResponder: () => true,
      onPanResponderGrant: () => {
        this.state.pan.setOffset({
          x: this._val.x,
          y: this._val.y
        });
        this.state.pan.setValue({ x: 0, y: 0 });
      },

      onStartShouldSetPanResponderCapture: () => true,
      // eslint-disable-next-line no-dupe-keys
      onPanResponderGrant: () => {
        this.props.onPressIn();
        this.state.pan.setOffset({
          x: this._val.x,
          y: this._val.y
        });

        this.state.pan.setValue({ x: 0, y: 0 });
      },

      onPanResponderMove: (e, gesture) => {
        const postion = this.state.pan.__getValue();
        if (postion.x > -65) {
          this.props.moveText(this.state.pan);
          Animated.event(
            [null, { dx: this.state.pan.x, dy: new Animated.Value(0) }],
            { useNativeDriver: false }
          )(e, gesture);
        } else {
          this.props.onPressOut();
          this.state.pan.setValue({ x: 0, y: 0 });
        }
      },

      onPanResponderTerminationRequest: () => true,
      onPanResponderRelease: () => {
        this.props.onPressOut();

        Animated.spring(this.state.pan, {
          toValue: { x: 0, y: 0 },
          useNativeDriver: false
        }).start();
      }
    });
  }

  // eslint-disable-next-line class-methods-use-this
  isDropArea(gesture) {
    return gesture.moveY < 200;
  }

  // eslint-disable-next-line react/sort-comp
  render() {
    return (
      <View style={{ alignItems: 'center' }}>{this.renderDraggable()}</View>
    );
  }

  // eslint-disable-next-line consistent-return
  renderDraggable() {
    const panStyle = {
      transform: this.state.pan.getTranslateTransform()
    };
    if (this.state.showDraggable) {
      return (
        <Animated.View {...this.panResponder.panHandlers} style={panStyle}>
          {this.props.children}
        </Animated.View>
      );
    }
  }
}
