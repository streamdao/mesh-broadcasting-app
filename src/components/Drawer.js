import React, { Component } from 'react';
import {
  Container,
  ListItem,
  Body,
  Left,
  Button,
  Icon,
  Thumbnail,
  BackAndroid
} from 'native-base';
import {
  View, Text, StyleSheet, NativeModules, Alert
} from 'react-native';
import { connect } from 'react-redux';
import { closeMenu, openAdministrativePanel, isAdministrative } from '../store/aplication/aplicationAction';
import { images } from '../utils/constans';
import { toast } from '../utils/utils';

/**
 * view of the menu drawer
 * @class HeaderComponent
 * @extends {Component}
 *
 */
class DrawerComponent extends Component {
  constructor() {
    super();
    this.state = {
      clicks: 0,
    };
  }

  componentDidMount = () => {
    this.props.isAdministrative();
  };


  handleChange = (view) => {
    const { navigation } = this.props;
    navigation.navigate(view);
  };

  onclickCounter = () => {
    const count = this.state.clicks + 1;
    this.setState({
      clicks: count
    });

    if (this.state.clicks === 8) {
      this.props.openAdministrativePanel(() => {
        toast('great! you are now an administrator');
      });
    }
  }

  render() {
    const { screenProps, navigation } = this.props;
    return (
      <Container>
        <View style={styles.headerDrawer}>
          <Thumbnail
            style={{ width: '100%', height: '100%' }}
            source={images.logo.url}
          />

          {this.props.user.picture && (
            <Thumbnail
              source={{
                uri: `${this.props.user.picture}?${new Date().getDate()}`,
                cache: 'force-cache'
              }}
              style={styles.imageStyle}
            />
          )}
          {!this.props.user.picture && (
            <Thumbnail source={images.noPhoto.url} style={styles.imageStyle} />
          )}
          <Text style={styles.textTitle}>{this.props.user.name}</Text>
        </View>
        <View style={{ flex: 1 }}>
          <ListItem itemDivider>
            <Text>Locha Mesh</Text>
          </ListItem>
          <ListItem icon button onPress={() => this.handleChange('contacts')}>
            <Left>
              <Button style={{ backgroundColor: '#ef6c00' }}>
                <Icon active name="person" />
              </Button>
            </Left>
            <Body>
              <Text>{screenProps.t('Drawer:contacts')}</Text>
            </Body>
          </ListItem>

          <ListItem icon button onPress={() => this.handleChange('config')}>
            <Left>
              <Button style={{ backgroundColor: '#ef6c00' }}>
                <Icon active name="settings" />
              </Button>
            </Left>
            <Body>
              <Text>{screenProps.t('Drawer:setting')}</Text>
            </Body>
          </ListItem>

          {this.props.administrative
            && (
              <>
                <ListItem itemDivider>
                  <Text>{screenProps.t('Drawer:listName')}</Text>
                </ListItem>

                <ListItem icon button onPress={() => this.handleChange('administrative')}>
                  <Left>
                    <Button style={{ backgroundColor: '#ef6c00' }}>
                      <Icon type="MaterialIcons" active name="developer-mode" />
                    </Button>
                  </Left>
                  <Body>
                    <Text>Admin settings</Text>
                  </Body>
                </ListItem>
              </>
            )}
        </View>

        <View style={{ height: 60, alignItems: 'center', justifyContent: 'center' }}>
          <Button
            onPress={this.onclickCounter}
            disabled={this.props.administrative}
            transparent
            style={{
              width: '100%',
              alignItems: 'center',
              justifyContent: 'center'
            }}
          >
            <Text>
              {`${screenProps.t('Drawer:version')} ${NativeModules.RNDeviceInfo.VersionInfo}`}
            </Text>
          </Button>
        </View>
      </Container>
    );
  }
}

const mapStateToProps = (state) => ({
  menu: state.aplication.menu,
  user: state.config,
  administrative: state.aplication.administrative,

});

export default connect(
  mapStateToProps,
  {
    closeMenu,
    isAdministrative,
    openAdministrativePanel
  }
)(DrawerComponent);

const styles = StyleSheet.create({
  headerDrawer: {
    height: '22%',
    backgroundColor: '#FAB300',
    justifyContent: 'center',
    alignItems: 'center'
  },
  imageStyle: {
    height: 60,
    width: 60,
    // borderRadius: 100,
    position: 'absolute',
    backgroundColor: 'red',
    right: '10%',
    top: '20%'
  },
  textTitle: {
    position: 'absolute',
    color: '#fff',
    fontSize: 30,
    fontWeight: '700',
    bottom: '15%',
    left: 15
  }
});
