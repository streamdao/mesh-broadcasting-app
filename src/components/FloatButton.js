import React, { Component } from 'react';
import {
  Text, TouchableOpacity, StyleSheet, View
} from 'react-native';
/**
 *
 *
 * @export
 * @class FloatButtons
 * @description reusable component is a floating button
 * @extends {Component}
 */
export default class FloatButtons extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <View>
        <TouchableOpacity
          onPress={() => this.props.add()}
          style={styles.floatButton}
        >
          {!this.props.icon && <Text testID="addIcon" style={styles.textButton}> + </Text>}
          {this.props.icon && this.props.icon}
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  floatButton: {
    width: 50,
    height: 50,
    position: 'absolute',
    backgroundColor: '#fbc233',
    right: 20,
    bottom: 15,
    borderRadius: 100,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: 'black',
    shadowOffset: {
      width: 0,
      height: 10
    },
    shadowOpacity: 0.8,
    shadowRadius: 100,

    elevation: 3
  },

  textButton: {
    fontSize: 23,
    color: 'black',
    marginRight: 6
  }
});
