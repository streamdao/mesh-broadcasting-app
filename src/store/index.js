import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import logger from 'redux-logger';
import { rootReducer } from './root';

let middleWare;
if (process.env.JEST_WORKER_ID) {
  middleWare = applyMiddleware(ReduxThunk)(createStore);
} else {
  middleWare = applyMiddleware(ReduxThunk)(createStore);
}


export default middleWare(rootReducer);
