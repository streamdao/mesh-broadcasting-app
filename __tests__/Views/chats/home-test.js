import '../../../__Mocks__';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { navigationPops } from '../../components/heder-test';
import Home from '../../../src/views/home';
import store from '../../../src/store';


const screenProps = {
  t: (data) => data
};

const mockMessage = [{
  fromUID: '16Uiu2HAm9JHK1iG6g2nLNw2fmAWerGiVoKMnjcvEs6pJVnd6STPC',
  id: 'ad7ae2f9dc672c55095e60f2e8367393261088972af35c49c007be0c642fda24',
  msg: 'Hola',
  msgID: 'ad7ae2f9dc672c55095e60f2e8367393261088972af35c49c007be0c642fda24',
  name: undefined,
  shippingTime: 1602013395803,
  status: 'pending',
  time: 1602013395803,
  timestamp: 1602013395786,
  toUID: 'broadcast',
  type: 1
}];

const mockChat = [{
  fromUID: '02ce1eeb78647383eac7998e929aa72240cc3d908f682f0375738330ab128bdea7',
  messages: { ...mockMessage },
  queue: [],
  toUID: 'broadcast'
}];


const mockContact = [{
  hashUID: '1ed4091506f77825bccb4dadcf89f1b62ff3257d2afc3886bf859c0330120a27',
  name: 'Luis',
  nodeAddress: '/ip4/85.53.137.144/tcp/4444',
  picture: null,
  uid: '16Uiu2HAm2BXA1TTUE7vvSu3U4YURxb4FwTYjfrwnZN618jrYBpNS'
}];

describe('test home componente', () => {
  const wrapper = shallow(
    <Home
      store={store}
      screenProps={screenProps}
      navigation={navigationPops}
    />
  ).childAt(0).dive();


  test('render home component', () => {
    expect(wrapper).toMatchSnapshot();
  });


  test('set props contacts', () => {
    wrapper.setProps({
      contacts: mockContact
    });
    expect(wrapper.instance().props.contacts).toBeDefined();
  });

  test('set props chat', () => {
    wrapper.setProps({
      chats: mockChat
    });
    expect(wrapper.instance().props.chats).toBeDefined();
  });


  test('selectedChat function', () => {
    wrapper.instance().selectedChat(mockContact[0], mockChat[0]);
    expect(navigationPops.navigate.mock.calls.length).toBe(1);
  });
});
