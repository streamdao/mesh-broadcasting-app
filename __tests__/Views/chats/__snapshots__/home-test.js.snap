// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home componente render home component 1`] = `
<Styled(Container)>
  <Connect(HeaderComponent)
    back={[Function]}
    broadcast={false}
    chats={Array []}
    contacts={Array []}
    delete={[Function]}
    deleteChat={[Function]}
    navigation={
      Object {
        "closeDrawer": [MockFunction],
        "dismiss": [MockFunction],
        "goBack": [MockFunction],
        "navigate": [MockFunction],
        "openDrawer": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "pop": [MockFunction],
        "popToTop": [MockFunction],
        "push": [MockFunction],
        "replace": [MockFunction],
        "reset": [MockFunction],
        "setParams": [MockFunction],
        "state": Object {
          "key": "id-1581513550398-0",
          "routeName": "initial",
        },
        "toggleDrawer": [MockFunction],
      }
    }
    screenProps={
      Object {
        "t": [Function],
      }
    }
    search={[Function]}
    selected={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
  <Styled(Content) />
  <FloatButtons
    add={[Function]}
    icon={
      <Styled(Icon)
        name="message"
        style={
          Object {
            "color": "#f5f5f5",
            "fontSize": 24,
          }
        }
        type="MaterialIcons"
      />
    }
  />
</Styled(Container)>
`;
